databroker/core.py

```diff
@@ -1024,18 +1039,12 @@ def _load(self):
                 fillers=OrderedDict(self.fillers),
                 transforms=OrderedDict(self._transforms),
                 metadata=metadata)
-            self._entries[stream_name] = StreamEntry(
-                name=stream_name,
-                description={},  # TODO
-                driver='databroker.core.BlueskyEventStream',
-                direct_access='forbid',
-                args=args,
-                cache=None,  # What does this do?
-                metadata=metadata,
-                catalog_dir=None,
-                getenv=True,
-                getshell=True,
-                catalog=self)
+
+            entries[stream_name] = functools.partial(wrapper, stream_name,
+                                                     descriptors, metadata, args)
+
+        self._entries = LazyMap(entries)
```

Comment:
We can override

```py
def _make_entries_container(self):
    return LazyMap()
```

to make `self._entires` a `LazyMap` to begin with.

Re-setting an attribute like this during the lifecycle of an object can create unexpected side-effects. As discussed with @gwbischof and @tacaswell toward the end of the day yesterday, I think we need to relent on making `LazyMap` mutable. However, we can stay true to the spirit of "make it act like a dict" and "keep cognitive load for the recipient of the dict low" by making a special setter method

```py
lazy_map = LazyMap(a=lambda: 1, b=lambda: 2)
lazy_map.add(c=lambda: 3)
```

The rationale for making a new method is to avoid the ambiguity of "Is it `lazy_map['c'] = lambda: 3` or `lazy_map['c'] = 3`?" If someone is using the special method `.add(...)` they know they are doing something outside of ordinary dict behavior.

We do not need to support updating existing values, and that would add some friction, so I think that trying to `add` a key that exists should blow up.